### PR TITLE
rnafold and SHAPE for multiple sequence

### PR DESCRIPTION
**Pull request by eteriSokhoyan**
running rnafold + SHAPE for multiple sequences.

### CODE REVIEW

File: tools/rna_tools/vienna_rna/rnafold_SHAPE.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        else:
+            with open(id_line +'.tmp', "a") as clFile:
+                clFile.write(line + "\n")
+
+
+input_file = sys.stdin
+
+for record in SeqIO.parse(input_file, "fasta"):
+    seq = ">{}\n{}".format(record.id,record.seq)
+    cmd =  'echo -e \"' + seq +'\" | RNAfold ' + "--shape=" + record.id + '.tmp ' + params_no_shape
```
Comment (bgruening) [inline tools/rna_tools/vienna_rna/rnafold_SHAPE.py]: I think using https://docs.python.org/2/library/subprocess.html#subprocess.call here with stdin is more secure.

File: tools/rna_tools/vienna_rna/rnafold_SHAPE.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+shape_file = shape_file.replace('--shape=', '')
+
+params_no_shape  =  " ".join(str(x) for x in param_list_no_shape)
+
+pattern = re.compile("^>.*$")
+id_line = ""
+with open(shape_file, 'r') as f:
+    content = f.read()
+    lines = content.split('\n')
+    for i in range(0, len(lines)):
```
Comment (bgruening) [inline tools/rna_tools/vienna_rna/rnafold_SHAPE.py]: `for line in lines:`

File: tools/rna_tools/vienna_rna/rnafold_SHAPE.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+### single sequence.
+
+import os
+import sys
+from os import system
+from Bio import SeqIO
+import re
+
+def sh(script):
+    system("bash -c '%s'" % script)
```
Comment (bgruening) [inline tools/rna_tools/vienna_rna/rnafold_SHAPE.py]: please use subprocess if you can

**Comment by bgruening** [comment]
This looks great! Could you add one test that is actually using this functionality.

File: tools/rna_tools/vienna_rna/rnafold.xml [hunk 4]
```diff
@@ -292,19 +299,28 @@
...
+
         <test>
             <param name="select_fasta" value="false" />
             <param name="input_sequence" value="TGGGAATTAGCTCAAATGGTAGAGCGCTCGCTTAGCATGTGAGAGGTAGTGGGATCGATGCCCACATTCTCCA"/>
             <output name="out_file1" file="rnafold_result3.txt"/>
         </test>
+        <test>
+            <param name="select_fasta" value="true" />
+            <param name="fasta_input" value="sample_3.fa"/>
+            <conditional name="shapeOption">
```
Comment (mmiladi) [inline tools/rna_tools/vienna_rna/rnafold.xml]: @bgruening If this test is looking fine, then the entire update looks good to me.

**Comment by bgruening** [comment]
Thanks @eteriSokhoyan!

**Pull request merged**